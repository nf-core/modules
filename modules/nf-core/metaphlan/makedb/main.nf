process METAPHLAN_MAKEDB {
    label 'process_medium'

    conda "bioconda::metaphlan=4.0.6"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/metaphlan:4.0.6--pyhca03a8a_0' :
        'quay.io/biocontainers/metaphlan:4.0.6--pyhca03a8a_0' }"

    output:
    path "metaphlan_db_latest"      , emit: db
    path "versions.yml"         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''

    """
    metaphlan \\
        --install \\
        --nproc $task.cpus \\
        --bowtie2db metaphlan_db_latest \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        metaphlan: \$(metaphlan --version 2>&1 | awk '{print \$3}')
    END_VERSIONS
    """
    stub:
    """
    mkdir metaphlan_db_latest
    touch metaphlan_db_latest/mpa_latest
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.1.bt2
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.2.bt2
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.3.bt2
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.4.bt2
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.fna.bz2
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.md5
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.pkl
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.rev.1.bt2
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.rev.2.bt2
    touch metaphlan_db_latest/mpa_v31_CHOCOPhlAn_201901.tar

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        metaphlan: \$(metaphlan --version 2>&1 | awk '{print \$3}')
    END_VERSIONS
    """
}


