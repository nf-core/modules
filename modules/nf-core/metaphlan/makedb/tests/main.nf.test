
nextflow_process {

    name "Test Process METAPHLAN_MAKEDB"
    script "../main.nf"
    process "METAPHLAN_MAKEDB"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "metaphlan"
    tag "metaphlan/makedb"

    test("test-metaphlan-makedb") {

        when {
            process {
                """
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.db[0].toString()).listFiles().collect { it.name }.toSorted(), // A mix of empty and non-empty files
                    process.out.versions
                    ).match()
                }
            )
        }
    }

    test("test-metaphlan-makedb-stub") {
        options '-stub'
        when {
            process {
                """
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
