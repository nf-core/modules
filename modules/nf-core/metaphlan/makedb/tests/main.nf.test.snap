{
    "test-metaphlan-makedb": {
        "content": [
            [
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103.1.bt2l",
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103.2.bt2l",
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103.3.bt2l",
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103.4.bt2l",
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103.fna.bz2",
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103.pkl",
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103.rev.1.bt2l",
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103.rev.2.bt2l",
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103_VINFO.csv",
                "mpa_vJan21_TOY_CHOCOPhlAnSGB_202103_VSG.fna"
            ],
            [
                "versions.yml:md5,5e0a5d647d68d6ad9d57c6138ddf7d45"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T20:55:39.743819"
    },
    "test-metaphlan-makedb-stub": {
        "content": [
            {
                "0": [
                    [
                        "mpa_latest:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.1.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.2.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.3.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.4.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.fna.bz2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.md5:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.pkl:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.rev.1.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.rev.2.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.tar:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,5e0a5d647d68d6ad9d57c6138ddf7d45"
                ],
                "db": [
                    [
                        "mpa_latest:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.1.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.2.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.3.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.4.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.fna.bz2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.md5:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.pkl:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.rev.1.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.rev.2.bt2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "mpa_v31_CHOCOPhlAn_201901.tar:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5e0a5d647d68d6ad9d57c6138ddf7d45"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-30T20:55:46.394376"
    }
}